ocaml/parsing/parser.mly

```diff
@@ -4536,9 +4536,26 @@ atomic_type:
       { [] }
   | ty = atomic_type
       { [ty] }
-  | LPAREN tys = separated_nontrivial_llist(COMMA, core_type) RPAREN
+  | LPAREN
+    tys = separated_nontrivial_llist(COMMA, one_type_parameter_of_several)
+    RPAREN
       { tys }
 
+(* Layout annotations on type expressions typically require parens, as in [('a :
+   float64)].  But this is unnecessary when the type expression is used as the
+   parameter of a tconstr with more than one argument, as in [(int, 'b :
+   float64) t].  Further, it would be confusing to require parens there because
+   they are not required in that position in type declarations.  So, we have a
+   special case. *)
```

Comment:
Nit: I think this comment is just as informative if you leave off the last two sentences. I don't have a strong preference.

Reply:
Done.